test/sqllogictest/relation-cse.slt

```diff
@@ -759,59 +721,92 @@ SELECT * FROM (SELECT a2.f1 AS f1 FROM t1 AS a1 LEFT JOIN t1 AS a2 USING (f1)) W
 ----
 %0 = Let l0 =
 | Get materialize.public.t1 (u1)
-| Filter !(isnull(#0))
+| Filter (#0 = 1)
+| Project (#1)
 
 %1 =
 | Get %0 (l0)
-| ArrangeBy (#0)
+| ArrangeBy ()
 
-%2 = Let l1 =
-| Join %1 %0 (= #0 #2)
-| | implementation = Differential %0 %1.(#0)
-| | demand = (#0, #1)
+%2 =
+| Get %0 (l0)
+| Project ()
 
-%3 =
-| Get %2 (l1)
-| Project (#0, #1, #0, #3)
+%3 = Let l1 =
+| Join %1 %2
+| | implementation = Differential %2 %1.()
```

Comment:
This plan (and a second one like it in the same file) is now significantly longer than it was before.

The plan itself is not necessarily bad though. It is now a union of
* cross join `Filter{ t1, [#0 = 1]}` to itself.
* cross join `Filter{ t1, [#0 = 2]}` to itself.
Which is smaller than a single join `t1` to itself on `#0`